pkg/operator/ceph/cluster/controller.go

```diff
@@ -276,6 +278,29 @@ func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, *cephCluster, errors.Wrapf(err, "failed to reconcile cluster %q", cephCluster.Name)
 	}
 
+	if cephCluster.Spec.Storage.Store.UpdateStore == osd.OSDStoreUpdateConfirmation {
+		cluster, ok := r.clusterController.clusterMap[request.Namespace]
+		if !ok {
+			return reconcile.Result{}, *cephCluster, errors.Wrap(err, "failed to get cluster map")
+		}
+
+		replaceConfig, err := osd.GetOSDReplaceConfigMap(r.context, cluster.ClusterInfo)
+		if err != nil {
+			return reconcile.Result{}, *cephCluster, errors.Wrap(err, "failed to get OSD replace config")
+		}
+
+		// Set progressing status if OSD migration is pending and reconcile after an interval if PGs are not healthy
+		if replaceConfig != nil && len(replaceConfig.OSDs) > 0 {
+			message := fmt.Sprintf("%d osds awaiting migration to %q", len(replaceConfig.OSDs), cephCluster.Spec.Storage.Store.Type)
+			opcontroller.UpdateCondition(cluster.ClusterInfo.Context, r.context, request.NamespacedName, k8sutil.ObservedGenerationNotAvailable, cephv1.ConditionProgressing, v1.ConditionTrue, cephv1.ClusterProgressingReason, message)
+			if !replaceConfig.IsPgHealthy {
```

Comment:
If PGs are not healthy, the OSD replacement won't start, right? And while the OSD replacement is in progress, we expect the PGs to be unhealthy until the replacement is completed and the data is backfilled. Why do we need to check the PG health here? Is the purpose just to requeue the reconcile to wait for the PGs to be healthy? I'm thinking the PG health check only needs to be at the moment that we decide whether to start a new OSD replacement.

Reply:
>>If PGs are not healthy, the OSD replacement won't start, right?
yes

>>Is the purpose just to requeue the reconcile to wait for the PGs to be healthy
That was the idea. To have `IsPgHealthy` in the config map and control the reconcile strategy based on that. But I think we don't need that now. So removing `IsPgHealthy` from the config map. Also moving the pg health check code where we decide whether to start a new osd replacement.

Reply:
Ignore above message. Now we can upgrade clusters that haven't migrated yet. Marking this comment as  resolved for now.